clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -140,17 +172,34 @@ public long metadataExpireMs() {
     }
 
     /**
-     * Request an update of the current cluster metadata info, return the current updateVersion before the update
+     * Request an update of the current cluster metadata info, permitting backoff based on the number of
+     * equivalent metadata responses, which indicates that responses did not make progress and may be stale.
+     * @param permitBackoffOnEquivalentResponses Whether to permit backoff when consecutive responses are equivalent.
+     *                                           This should be set to <i>true</i> in situations where the update is
+     *                                           being requested to retry an operation, such as when the leader has
+     *                                           changed. It should be set to <i>false</i> in situations where new
+     *                                           metadata is being requested, such as adding a topic to a subscription.
+     *                                           In situations where it's not clear, it's best to use <i>false</i>.
+     * @return The current updateVersion before the update
      */
-    public synchronized int requestUpdate() {
+    public synchronized int requestUpdate(final boolean permitBackoffOnEquivalentResponses) {
         this.needFullUpdate = true;
+        if (!permitBackoffOnEquivalentResponses) {
```

Comment:
Since we only take action when permitBackoffOnEquivalentResponses is false, would it be more intuitive to pass in the negation of that as sth like resetEquivalentResponseCount?

Reply:
I think the negation is a little clearer.